main/src/ca/uwaterloo/flix/language/phase/Parser2.scala

```diff
@@ -2533,9 +2533,11 @@ object Parser2 {
       Type.ttype()
 
       // NewStruct, NewObject, or InvokeConstructor?
-      if (at(TokenKind.CurlyL) && (nth(1) == TokenKind.NameLowerCase || (nth(1) == TokenKind.CurlyR && nth(2) == TokenKind.At))) {
-        // case 2: new Struct {field1 = expr1, field2 = expr2, ...} @ region
-        //     or: new Struct {} @ region
+      if (at(TokenKind.At)) {
+        // case 2: new Struct @ rc {field1 = expr1, field2 = expr2, ...}
+        //     or: new Struct @ rc {}
+        expect(TokenKind.At, SyntacticContext.Expr.OtherExpr)
+        expression()
         zeroOrMore(
```

Comment:
if zeroOrMore doesnt see the left delimiter it will just return 0 (no error or anything. This means (I think) that `new Struct @ r |> ???` parsers correctly?

Reply:
maybe a a negative test somewhere